unitTests/ui/CategoryFacet/CategoryFacetTest.ts

```diff
@@ -27,6 +27,105 @@ export function CategoryFacetTest() {
+          master = Mock.advancedComponentSetup<CategoryFacet>(
+            CategoryFacet,
+            new Mock.AdvancedComponentSetupOptions(
+              undefined,
+              {
+                field: masterFacetField
+              },
+              env => env.withLiveQueryStateModel()
+            )
+          );
+
+          test = Mock.advancedComponentSetup<CategoryFacet>(
+            CategoryFacet,
+            new Mock.AdvancedComponentSetupOptions(
+              undefined,
+              {
+                field: dependentFacetField,
+                dependsOn: masterFacetField
+              },
+              env => env.withQueryStateModel(master.cmp.queryStateModel)
+            )
+          );
+        });
+
+        it('adds the coveo-facet-dependent class to the dependent facet', () => {
+          Simulate.query(test.env, simulateQueryDataDependant);
+          expect($$(test.cmp.element).hasClass('coveo-facet-dependent')).toBe(true);
+        });
+
+        it('does not add the coveo-facet-dependent class to the master facet', () => {
+          Simulate.query(master.env, simulateQueryDataMaster);
+          expect($$(master.cmp.element).hasClass('coveo-facet-dependent')).toBe(false);
+        });
+
+        it(`when the master facet has a selected value,
+        it removes the coveo-facet-dependent class from the dependent facet`, () => {
+          Simulate.query(master.env, simulateQueryDataMaster);
+          master.cmp.selectValue('value9');
+          expect($$(test.cmp.element).hasClass('.coveo-facet-dependent')).toBe(false);
+        });
+
+        it(`when the master facet has a selected value,
+        it adds the coveo-category-facet-non-empty-path class to the master facet`, () => {
+          Simulate.query(master.env, simulateQueryDataMaster);
+          master.cmp.selectValue('value9');
+          expect($$(master.cmp.element).hasClass('.coveo-category-facet-non-empty-path')).toBe(false);
+        });
+
+        describe('when the master and the depeandents have a selected value', () => {
+          function triggerStateChangeOnDependentFacet() {
+            rebindDependentFacetStateChangeListeners();
+            $$(test.env.root).trigger('change', { attributes: {} });
+          }
+
+          function rebindDependentFacetStateChangeListeners() {
+            test.cmp.queryStateModel.getEventName = name => name;
+            test.cmp['initQueryStateEvents']();
+          }
+
+          beforeEach(() => {
+            Simulate.query(master.env, simulateQueryDataMaster);
+            master.cmp.selectValue('value9');
+            Simulate.query(test.env, simulateQueryDataDependant);
+            test.cmp.selectValue('value5');
+          });
+
+          it(`when resetting the master facet,
+          it reset the dependent facet`, () => {
+            expect(test.cmp.activePath.length).toBe(1);
+            master.cmp.deselectCurrentValue();
+            triggerStateChangeOnDependentFacet();
```

Comment:
🤔 Why do you need to trigger that manually here ? Aren't you creating a `liveQueryStateModel()` in your setup, meaning that the events should be launched properly in your tests, no ?

Reply:
As stated before, you're right but i need to call `test.cmp['initQueryStateEvents']();`. I'll check why I need to recall it as it should be called the initialization of my component